src/components/componentDiff/ComponentDiffTab.js

```diff
@@ -219,6 +221,90 @@ const ComponentVersionDiff = ({
     rightVersion: rightVersion,
   })
 
+  const determineMainSourceForComponent = (component, absentOk = true) => {
+    if (!component.sources || component.sources.length === 0) {
+      if (absentOk) {
+        return null
+      } else {
+        throw new Error(`No sources defined for the component: ${JSON.stringify(component)}`)
+      }
+    }
+  
+    const mainSource = component.sources.find(source => {
+      const label = source.labels?.find(l => l.name === knownLabelNames.mainSource)
+      return label && label.value && label.value['repository-classification'] === 'main'
+    })
+  
+    if (mainSource) {
+      return mainSource.access?.repoUrl || null
+    }
+  
+    if (absentOk) {
+      return component.sources[0]?.access?.repoUrl || null
+    } else {
+      throw new Error(`No main source could be identified for the component: ${JSON.stringify(component)}`)
+    }
+  }
+
+  const prefixKnownResourceTypes = (resource, component) => {
+    const shortComponentName = shortenComponentName(component.name)
+
+    if (resource.type === knownResourceTypes.OCI_IMAGE) {
+      return `[Docker] ${shortComponentName}/${resource.name}`
+    }
+
+    if (resource.type === knownResourceTypes.HELM_CHART_V1) {
+      const chartName = component.name.split('/').slice(-2, -1)[0]
+      return `[Docker] ${shortComponentName}/charts/${chartName}/${resource.name}`
+    }
+
+    return `${shortComponentName}/${resource.name}`
+  }
+
+  const handleExportDiff = () => {
+    const { addedResources, removedResources } = (diff.components_changed || []).reduce(
```

Comment:
can't we just return the json-document we received from delivery-service earlier?